### PR TITLE
Update README in preparation for 0.9.0 release

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

N/A

 # Rationale for this change

Update README to reflect recent improvements. 

# What changes are included in this PR?

- General update of README
- Add roadmap (was previous in datafusion repo)

# Are there any user-facing changes?

No

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,63 +17,91 @@
...
-that, refer to the [Getting Started Guide](ballista/rust/client/README.md).
+## Roadmap
+
+There is an excellent discussion in https://github.com/apache/arrow-ballista/issues/30 about the future of the project
+and we encourage you to participate and add your feedback there if you are interested in using or contributing to
+Ballista.
+
+The current focus is on the following items:
+
+- Make production ready
```
Comment (andygrove) [inline README.md:72]: I copied this section from https://github.com/apache/arrow-datafusion/pull/3493

File: README.md [hunk 2]
```diff
@@ -17,63 +17,91 @@
...
+    - Schedule restart
+  - Improve monitoring and logging
+  - Auto scaling support
+  - Better configuration management
+- All-at-once job task scheduling
+- Shuffle improvement
+  - Shuffle memory control
+  - Improve shuffle IO to avoid producing too many files
+  - Support sort-based shuffle
+  - Support range partition
```
Comment (Dandandan) [inline README.md:88]:
```suggestion
  - Support broadcast shuffle
```

**Comment by andygrove** [comment]
@mingmwang @avantgardnerio Let me know if there is anything you would like to add

**Comment by andygrove** [comment]
I'll go ahead and merge this since there have been no objections in 10 days. Feel free to PR to make additional changes!

**Pull request merged**